src/webhook/validation/deprecation.go

```diff
@@ -56,6 +104,19 @@ func warnIfDeprecatedIsUsed(dynakube *dynatracev1beta1.DynaKube, newAnnotation s
 	return ""
 }
 
+func warnIfDeprecatedFFisUsed(dynakube *dynatracev1beta1.DynaKube, annotation string) string {
+	_, hasDeprecatedFlag := dynakube.Annotations[annotation]
+	if hasDeprecatedFlag {
+		return deprecationWarning(annotation)
+	}
+
+	return ""
+}
+
 func deprecatedAnnotationWarning(newAnnotation string, deprecatedAnnotation string) string {
 	return fmt.Sprintf("annotation '%s' is deprecated, use '%s' instead", deprecatedAnnotation, newAnnotation)
 }
+
+func deprecationWarning(annotation string) string {
+	return fmt.Sprintf("feature flag '%s' is deprecated and will be removed in the future as the feature will be enabled by default", annotation)
```

Comment:
this message would also appear if the ff is used and already enabled, leading to a confusing warning.

I think it's enough to just warn that the ff will be removed in the future and we'll communicate the change in default value in the release notes